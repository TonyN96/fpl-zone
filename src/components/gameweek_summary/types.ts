export interface SummaryStatType {
  label: string;
  teamCode?: number;
  playerName?: string;
  value?: string | number;
}
