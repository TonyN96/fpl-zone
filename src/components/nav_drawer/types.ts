export interface MenuItemType {
  id: string;
  label: string;
  type: "nav" | "auth";
  onClick: () => void;
}
