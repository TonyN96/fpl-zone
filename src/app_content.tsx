import React from "react";
import { useQuery } from "react-query";
import { BrowserRouter as Router,Route, Routes } from "react-router-dom";
import { useMediaQuery } from "@mui/material";
import { getAllFixtures, getGameData } from "api/fpl_api_provider";
import {
  AccountPage,
  AnalysisPage,
  FixturesAndResultsPage,
  GameweekLivePage,
  LoginPage,
  MyTeamPage,
  RegisterPage,
  ResetPage
} from "pages";
import PrivateRoute from "private_route";
import { AppData } from "types";

import { Logout } from "components/authentication";
import { Notifier, Startup } from "components/layout";

export const AppDataContext = React.createContext<AppData | null>(null);

export default function AppContent(): JSX.Element {
  // Fetching game data which will be made available throughout the app via context provider
  const {
    data: gameData,
    isError: gameDataIsError,
    error: gameDataError,
    isLoading: gameDataIsLoading
  } = useQuery("game-data", getGameData);

  // Fetching fixture data which will be made available throughout the app via context provider
  const {
    data: fixtureData,
    isError: fixtureDataIsError,
    error: fixtureDataError,
    isLoading: fixtureDataIsLoading
  } = useQuery("all-fixtures", getAllFixtures);

  const isCompact = useMediaQuery("(max-width:1500px)");

  const isLoading = gameDataIsLoading || fixtureDataIsLoading;
  const isError = gameDataIsError || fixtureDataIsError;
  const appData = gameData && fixtureData ? { ...gameData, fixtures: fixtureData, isCompact } : null;

  if (isLoading) {
    // Display loading message if data is still being fetched
    return (
      <Startup>
        <Notifier />
      </Startup>
    );
  } else if (isError) {
    // Display error message if data fetch failed
    const error = gameDataError || fixtureDataError;
    const errorMessage = error instanceof Error ? `: ${error.message}` : ".";

    return (
      <Startup>
        <Notifier type='error' message={`An error has occured: ${errorMessage}`} />
      </Startup>
    );
  }

  // Otherwise, render the app
  return (
    <AppDataContext.Provider value={appData}>
      <Router>
        <Routes>
          <Route path='*' element={<LoginPage />} />
          <Route path='/' element={<LoginPage />} />
          <Route path='/login' element={<LoginPage />} />
          <Route path='/register' element={<RegisterPage />} />
          <Route path='/reset' element={<ResetPage />} />
          <Route path='/logout' element={<Logout />} />
          <Route path='/account' element={<PrivateRoute component={<AccountPage />} />} />
          <Route
            path='/gameweek-live'
            element={<PrivateRoute component={<GameweekLivePage />} />}
          />
          <Route path='/my-team' element={<PrivateRoute component={<MyTeamPage />} />} />
          <Route
            path='/fixtures-and-results'
            element={<PrivateRoute component={<FixturesAndResultsPage />} />}
          />
          <Route path='/analysis' element={<PrivateRoute component={<AnalysisPage />} />} />
        </Routes>
      </Router>
    </AppDataContext.Provider>
  );

}
