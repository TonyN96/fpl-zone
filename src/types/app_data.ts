import { Fixture, GameSettings, Gameweek, Phase, Player, PlayerStat, Position,Team } from "types";

export interface AppData {
  gameweeks: Gameweek[];
  gameSettings: GameSettings;
  phases: Phase[];
  teams: Team[];
  playerCount: number;
  players: Player[];
  playerStats: PlayerStat[];
  positions: Position[];
  fixtures: Fixture[];
  isCompact: boolean;
}
