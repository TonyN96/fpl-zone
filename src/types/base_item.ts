import { Player } from "./player";
import { Team } from "./team";

export type BaseItem = Player | Team;

export const isPlayer = (item: BaseItem): item is Player => {
  return "web_name" in item;
};
