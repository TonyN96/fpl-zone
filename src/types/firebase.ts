export interface FirebaseResponse {
  code: string;
  message: string;
  name: string;
}
