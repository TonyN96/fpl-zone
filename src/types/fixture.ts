export interface Fixture {
  id: number;
  code: number;
  team_h: number;
  team_h_score?: unknown;
  team_a: number;
  team_a_score?: unknown;
  event: number;
  finished: boolean;
  minutes: number;
  provisional_start_time: boolean;
  kickoff_time: Date;
  event_name: string;
  is_home: boolean;
  difficulty: number;
}
