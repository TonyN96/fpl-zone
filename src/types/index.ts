export * from "./app_data";
export * from "./fixture";
export * from "./game_data";
export * from "./game_settings";
export * from "./gameweek";
export * from "./league";
export * from "./lineup";
export * from "./phase";
export * from "./player";
export * from "./player_past";
export * from "./player_performance";
export * from "./player_stat";
export * from "./player_summary";
export * from "./position";
export * from "./region";
export * from "./team";
export * from "./team_data";
export * from "./team_history";
export * from "./team_picks";
export * from "./transfer";
