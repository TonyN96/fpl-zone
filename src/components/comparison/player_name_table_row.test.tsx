import React from "react";
import { render, screen } from "@testing-library/react";
import { AppDataContext } from "app_content";
import { mockAppData, mockPlayers } from "test/test_data";
import { Player } from "types";

import "@testing-library/jest-dom/extend-expect";

import { PlayerNameTableRow } from ".";

describe("Player name table row tests", () => {
  let players: Player[];

  const createComponent = (): JSX.Element => {
    return (
      <AppDataContext.Provider value={mockAppData}>
        <PlayerNameTableRow players={players} />
      </AppDataContext.Provider>
    );
  };

  it("Snapshot test", () => {
    players = mockPlayers.slice(0, 4);

    const { asFragment } = render(createComponent());
    expect(asFragment()).toMatchSnapshot();
  });

  it("Player's name displayed correctly", () => {
    render(createComponent());

    players.forEach((player) => {
      const playerNameContainer = screen.getByTestId(`player-name-row-${player.id}`);
      expect(playerNameContainer).toHaveTextContent(`${player.first_name} ${player.second_name}`);
    });
  });

  describe("Empty table cell", () => {
    it("displayed when less than 5 five players are selected", () => {
      players = mockPlayers.slice(0, 2);

      render(createComponent());

      expect(screen.getByTestId("empty-table-cell")).toBeInTheDocument();
    });

    it("not displayed when 5 five players are selected", () => {
      players = mockPlayers.slice(0, 5);

      render(createComponent());

      expect(screen.queryByTestId("empty-table-cell")).toBeNull();
    });
  });
});
