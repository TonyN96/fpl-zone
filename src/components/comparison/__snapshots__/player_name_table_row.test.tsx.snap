// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player name table row tests Snapshot test 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Name
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
  >
    Gerome Done
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
  >
    Ingmar Carwithan
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
  >
    Aharon Raccio
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-16rr9ae-MuiTypography-root"
  >
    Mick Poyzer
  </p>
</DocumentFragment>
`;
